synaptic_reconstruction/inference/postprocessing/vesicles.py

```diff
@@ -22,17 +22,30 @@ def close_holes(vesicle_segmentation, closing_iterations=4, min_size=0, verbose=
     return closed_segmentation
 
 
-def filter_zborder_objects(segmentation):
-    """Filter any object that touches the upper or lower z-border.
+def filter_border_objects(segmentation: np.ndarray, z_border_only: bool = False) -> np.ndarray:
```

Comment:
This function now filters objects from all borders (unless `z_border_only=True`, then it only filters in the depth axis)